### PR TITLE
[mailbox] E: System call mailbox_tag now in user space

### PR DESCRIPTION
**Pull request by SamuelVLC**

### CODE REVIEW

File: include/nanvix/kernel/kcall.h [hunk 1]
```diff
@@ -57,6 +57,7 @@
 #define NR_thread_exit 23   /** kernel_thread_exit()      */
 #define NR_thread_yield 24  /** kernel_thread_yield()     */
 #define NR_last_kcall 25    /** NR_SYSCALLS definer       */
+#define NR_mailbox_tag 26   /** kernel_mailbox_tag        */
```
Comment (ppenna) [inline include/nanvix/kernel/kcall.h:60]: `NR_mailbox_tag` should be 25 and `NR_last_kcall` should be 26.

File: src/kernel/kcall/ipc/boxtag.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#include <nanvix/kernel/lib.h>
+
+/*============================================================================*
+ * Public Functions                                                           *
+ *============================================================================*/
+
+/**
+ * @details Gets mailbox tag.
+ */
+
```
Comment (ppenna) [inline src/kernel/kcall/ipc/boxtag.c:20]: Remove this extra line.

File: src/kernel/kcall/ipc/boxtag.c [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+int kcall_mailbox_tag(int mbxid)
+{
+    // TODO: https://github.com/nanvix/microkernel/issues/400
+    // Return to test syscall.
+    return (mbxid);
```
Comment (ppenna) [inline src/kernel/kcall/ipc/boxtag.c:25]: Return `ENOTSUP` and use `UNUSED()` to avoid compiler warning.

File: src/libnanvix/src/ipc/mod.rs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+///
+/// - `image` - Image of the target process.
+///
+/// **Return**
+///
+/// Upon successful completion, the mailbox tag is returned.
+/// Upon failure, a negative error code is returned instead.
+///
+
+pub fn boxtag(mbxid: i32) -> i32 {
```
Comment (ppenna) [inline src/libnanvix/src/ipc/mod.rs:30]: Shall we name this `mailbox_tag`? So it matches the name of the kernel call.

File: src/servers/test/src/test.rs [hunk 5]
```diff
@@ -435,6 +436,17 @@ fn test_semget_call() -> bool {
     true
 }
 
+// Test systemcall for ipc module
+
```
Comment (ppenna) [inline src/servers/test/src/test.rs:440]: Remove this extra line.

File: src/servers/test/src/test.rs [hunk 6]
```diff
@@ -435,6 +436,17 @@ fn test_semget_call() -> bool {
...
+fn test_mailbox_tag() -> bool {
+    let mbxid: i32 = 1;
+    if mbxid != ipc::boxtag(mbxid){
+        return false;
```
Comment (ppenna) [inline src/servers/test/src/test.rs:444]: We should assert for `ENOTSUP`.

File: src/libnanvix/src/kcall/mod.rs [hunk 7]
```diff
@@ -47,6 +47,7 @@ pub enum KcallNumbers {
     Semget = 18,
     Semop = 19,
     Semctl = 20,
+    Boxtag = 26,
```
Comment (ppenna) [inline src/libnanvix/src/kcall/mod.rs:50]: This should match the number in the header.

**Review by ppenna — COMMENTED** [review]
Could you link the concerned issues in this PR?

**Comment by SamuelVLC** [comment]
> Could you link the concerned issues in this PR?

it is linked already.

File: include/nanvix/kernel/kcall.h (hunk 1 above)
Comment (SamuelVLC) [reply include/nanvix/kernel/kcall.h]: resolved in the new commit 419c7d6c36b229c637399ed488ae3402ae5cf953

File: src/kernel/kcall/ipc/boxtag.c (hunk 2 above)
Comment (SamuelVLC) [reply src/kernel/kcall/ipc/boxtag.c]: resolved in the new commit 419c7d6c36b229c637399ed488ae3402ae5cf953

File: src/kernel/kcall/ipc/boxtag.c (hunk 3 above)
Comment (SamuelVLC) [reply src/kernel/kcall/ipc/boxtag.c]: resolved in the new commit 419c7d6c36b229c637399ed488ae3402ae5cf953

File: src/libnanvix/src/ipc/mod.rs (hunk 4 above)
Comment (SamuelVLC) [reply src/libnanvix/src/ipc/mod.rs]: resolved in the new commit 419c7d6c36b229c637399ed488ae3402ae5cf953

**Comment by SamuelVLC** [comment]
All changes have been added as suggested.
See commit 419c7d6c36b229c637399ed488ae3402ae5cf953

File: src/libnanvix/src/ipc/mod.rs [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+/// **Parameters**
+///
+/// - `image` - Image of the target process.
+///
+/// **Return**
+///
+/// Upon successful completion, the mailbox tag is returned.
+/// Upon failure, a negative error code is returned instead.
+///
+
```
Comment (ppenna) [inline src/libnanvix/src/ipc/mod.rs:29]: Remove this extra line.

Comment (SamuelVLC) [reply src/libnanvix/src/ipc/mod.rs]: Done

**Review by ppenna — APPROVED** [review]

**Pull request merged**